m4 Doesn't Build With ARM Compiler

The m4 package fails to build with the ARM compiler with the error:
```
undefined reference to `__muloti4`
```

This is essentially the same error as mentioned in #4883, and the same fix will also work here.  (The ARM compiler is based on LLVM, so this isn't too surprising.)  Pull request will be coming shortly.

--- a/var/spack/repos/builtin/packages/m4/package.py
+++ b/var/spack/repos/builtin/packages/m4/package.py
@@ -34,7 +34,8 @@ class M4(AutotoolsPackage):
         spec = self.spec
         args = ['--enable-c++']
 
-        if spec.satisfies('%clang') and not spec.satisfies('platform=darwin'):
+        if (spec.satisfies('%clang') or spec.satisfies('%arm')) and not \
+           spec.satisfies('platform=darwin'):
             args.append('CFLAGS=-rtlib=compiler-rt')
 
         if spec.satisfies('%intel'):

